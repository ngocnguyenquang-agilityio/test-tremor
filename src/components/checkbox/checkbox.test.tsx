// Libs
import { cleanup, render, fireEvent, screen } from '@testing-library/react';

// Components
import CheckBox from './checkbox';

const mockOnClick = jest.fn();

const setup = () => {
  const props = {
    id: 'checkbox'
  };

  return render(<CheckBox data-testid="checkbox" onClick={mockOnClick} {...props} />);
};

describe('CheckBox component render', () => {
  afterEach(cleanup);

  test('should render CheckBox component correctly', () => {
    setup();

    const checkbox = screen.getByTestId('checkbox');

    expect(checkbox).toMatchSnapshot();
  });

  test('check CheckBox has attr checked', async () => {
    render(<CheckBox id='checkbox' data-testid="checkbox" defaultChecked={true}/>);

    const checkbox = screen.getByTestId('checkbox');
    
    expect(checkbox.hasAttribute('checked')).toBeTruthy();
  });

  test('check CheckBox has attr disabled', async () => {
    render(<CheckBox id='checkbox' data-testid="checkbox" disabled />);

    const checkbox = screen.getByTestId('checkbox');
    
    expect(checkbox.hasAttribute('disabled')).toBeTruthy();
  });

  test('should simulate onClick event on CheckBox', async() => {
    setup();

    const checkbox = screen.getByTestId('checkbox');

    expect(checkbox).toBeDefined();

    fireEvent.click(checkbox);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
