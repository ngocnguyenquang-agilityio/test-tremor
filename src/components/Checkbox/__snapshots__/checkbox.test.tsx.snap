// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox component render should render CheckBox component correctly 1`] = `
<input
  class="form-checkbox text-[#344767] h-5 w-5 border-[#dee2e6] rounded-md checked:bg-[length:90%_90%] border-[1px] checked:bg-[length:80%_80%] border-[1px] checked:bg-[length:50%_50%] border-[1px]"
  data-testid="checkbox"
  id="checkbox"
  type="checkbox"
/>
`;
